packages/flutter/lib/src/widgets/framework.dart

```diff
@@ -2185,6 +2185,18 @@ class StatefulElement extends ComponentElement {
   }
 
   @override
+  InheritedWidget inheritFromWidgetOfExactType(Type targetType) {
+    assert(() {
+      if (state._debugLifecycleState == _StateLifecycle.ready)
+        return true;
+      throw new FlutterError(
+        'inheritFromWidgetOfExactType($targetType) was called before ${_state.runtimeType}.initState() completed.'
```

Comment:
This string should be followed by a newline and then a paragraph explaining the reasoning and what to do about it